### ISSUE TITLE
KuCoinFutures `watch_balance`  AttributeError

### ISSUE THREAD

**Issue by BaoziCDR** [ccxt/ccxt#18996]
### Operating System

MacOS 13.4.1 (Apple M1)

### Programming Languages

Python 3.11.4

### CCXT Version

ccxtpro 4.0.74

### Description

When I call `watch_balance` of KuCoinFutures, it reports an error, here is the error message:

Traceback (most recent call last):
  File "/Users/jume/PycharmProjects/strategy_sdk_python/strategy-sdk-exchange/tests/main2.py", line 157, in <module>
    loop.run_until_complete(gogogo())
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/jume/PycharmProjects/strategy_sdk_python/strategy-sdk-exchange/tests/main2.py", line 152, in gogogo
    async for data in go:
  File "/Users/jume/PycharmProjects/strategy_sdk_python/strategy-sdk-exchange/tests/main2.py", line 144, in sub_kline
    data = await kucoin_pro.watch_balance()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jume/Library/Caches/pypoetry/virtualenvs/strategy-sdk-python-sLeCTyBG-py3.11/lib/python3.11/site-packages/ccxt/pro/kucoinfutures.py", line 499, in watch_balance
    url = await self.negotiate(True)
                ^^^^^^^^^^^^^^^^^^^^
  File "/Users/jume/Library/Caches/pypoetry/virtualenvs/strategy-sdk-python-sLeCTyBG-py3.11/lib/python3.11/site-packages/ccxt/pro/kucoinfutures.py", line 64, in negotiate
    urls[connectId] = self.spawn(self.negotiate_helper, privateChannel, params)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jume/Library/Caches/pypoetry/virtualenvs/strategy-sdk-python-sLeCTyBG-py3.11/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 321, in spawn
    task = self.asyncio_loop.create_task(method(*args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'create_task'

### Code

```
if __name__ == '__main__':
      async def sub_kline():
        kucoin_pro = ccxtpro.kucoinfutures()
        kucoin_pro.apiKey = "------"
        kucoin_pro.secret = "------"
        kucoin_pro.password = "------"
        while True:
            data = await kucoin_pro.watch_balance()
            yield data


    async def gogogo():
        go = sub_kline()
        if isinstance(go, AsyncGenerator):
            print("let's go")
            async for data in go:
                print(data)


    loop = asyncio.get_event_loop()
    loop.run_until_complete(gogogo())
```

**Comment by BaoziCDR** [comment]
It seems to be not calling `await self.load_markets()`
`
 async def watch_balance(self, params={}):
        """
        watch balance and get the amount of funds available for trading or funds locked in orders
        see https://docs.kucoin.com/futures/#account-balance-events
        :param dict [params]: extra parameters specific to the kucoinfutures api endpoint
        :returns dict: a `balance structure <https://docs.ccxt.com/en/latest/manual.html?#balance-structure>`
        """
+      await self.load_markets()     <----  Miss
        url = await self.negotiate(True)
        topic = '/contractAccount/wallet'
        request = {
            'privateChannel': True,
        }
        subscription = {
            'method': self.handle_balance_subscription,
        }
        messageHash = 'balance'
        return await self.subscribe(url, messageHash, topic, subscription, self.extend(request, params))
`

**Comment by carlosmiei** [comment]
Hello @BaoziCDR, thanks for reporting it, we will take a look as soon as possible.

**Comment by carlosmiei** [comment]
@BaoziCDR I just tested and seemed to be working, can you enable the verbose mode `exchange.verbose = True` and check if there's any error in the stream that we're missing?

```
 p krakenfutures watchBalance --sandbox          
Python v3.10.9
CCXT v4.0.75
krakenfutures.watchBalance()
{'BCH': {'free': 35.6328392246, 'total': 35.6328392246, 'used': 0},
 'BTC': {'free': 0.2071703309, 'total': 0.2071703309, 'used': 0},
 'ETH': {'free': 3.0445849013, 'total': 3.0445849013, 'used': 0},
 'EUR': {'free': 4699.2481203008, 'total': 4699.2481203008, 'used': 0},
 'GBP': {'free': 4139.4840547074, 'total': 4139.4840547074, 'used': 0},
 'LTC': {'free': 52.443885043, 'total': 52.443885043, 'used': 0},
 'USD': {'free': 0.0, 'total': 0.0, 'used': 0},
 'USDC': {'free': 5002.0008003201, 'total': 5002.0008003201, 'used': 0},
 'USDT': {'free': 5000.500050005, 'total': 5000.500050005, 'used': 0},
 'XRP': {'free': 12841.2563885251, 'total': 12841.2563885251, 'used': 0},
 'datetime': '2023-08-25T11:03:02.880Z',
 'info': {'account': '3b111acc-4fcc-45be-a622-57e611fe9f7f',
          'feed': 'balances_snapshot',
          'flex_futures': {'available_margin': 5008.148794545658,
                           'balance_value': 5004.46,
                           'collateral_value': 5004.46,
                           'cross': {'available_margin': 5008.148794545658,
                                     'balance_value': 5004.46,
                                     'collateral_value': 5004.46,
                                     'effective_leverage': 0.0029947001028128696,
                                     'initial_margin': 0.7000000000000001,
                                     'initial_margin_without_orders': 0.3,
                                     'maintenance_margin': 0.15,
                                     'margin_equity': 5008.848794545658,
                                     'pnl': 4.3887931719659985,
                                     'portfolio_value': 5008.848794545658,
                                     'total_unrealized': 4.388794545658429,
                                     'total_unrealized_as_margin': 4.38879454565843,
                                     'unrealized_funding': 1.373692431045e-06},
                           'currencies': {'ETH': {'available': 0.0,
                                                  'collateral_value': 0.0,
                                                  'conversion_spread': 0.25,
                                                  'haircut': 0.04,
                                                  'quantity': 0.0,
                                                  'value': 0.0},
                                          'EUR': {'available': 0.0,
                                                  'collateral_value': 0.0,
                                                  'conversion_spread': 0.25,
                                                  'haircut': 0.02,
                                                  'quantity': 0.0,
                                                  'value': 0.0},
                                          'GBP': {'available': 0.0,
                                                  'collateral_value': 0.0,
                                                  'conversion_spread': 0.25,
                                                  'haircut': 0.02,
                                                  'quantity': 0.0,
                                                  'value': 0.0},
                                          'USD': {'available': 5008.14879454566,
                                                  'collateral_value': 5004.45536590796,
                                                  'conversion_spread': 0.0,
                                                  'haircut': 0.0,
                                                  'quantity': 5004.45536590796,
                                                  'value': 5004.45536590796},
                                          'USDC': {'available': 0.0,
                                                   'collateral_value': 0.0,
                                                   'conversion_spread': 0.25,
                                                   'haircut': 0.01,
                                                   'quantity': 0.0,
                                                   'value': 0.0},
                                          'USDT': {'available': 0.0,
                                                   'collateral_value': 0.0,
                                                   'conversion_spread': 0.25,
                                                   'haircut': 0.05,
                                                   'quantity': 0.0,
                                                   'value': 0.0},
                                          'XBT': {'available': 0.0,
                                                  'collateral_value': 0.0,
                                                  'conversion_spread': 0.25,
                                                  'haircut': 0.02,
                                                  'quantity': 0.0,
                                                  'value': 0.0}},
                           'initial_margin': 0.7000000000000001,
                           'initial_margin_without_orders': 0.3,
                           'isolated': {},
                           'maintenance_margin': 0.15,
                           'margin_equity': 5008.848794545658,
                           'pnl': 4.3887931719659985,
                           'portfolio_value': 5008.848794545658,
                           'total_unrealized': 4.388794545658429,
                           'total_unrealized_as_margin': 4.38879454565843,
                           'unrealized_funding': 1.373692431045e-06},
          'futures': {'F-BCH:USD': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-BCH:USD',
                                    'pair': 'BCH/USD',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'BCH',
                                    'unrealized_funding': 0.0},
                      'F-ETH:EUR': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-ETH:EUR',
                                    'pair': 'ETH/EUR',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'EUR',
                                    'unrealized_funding': 0.0},
                      'F-ETH:USD': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-ETH:USD',
                                    'pair': 'ETH/USD',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'ETH',
                                    'unrealized_funding': 0.0},
                      'F-LTC:USD': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-LTC:USD',
                                    'pair': 'LTC/USD',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'LTC',
                                    'unrealized_funding': 0.0},
                      'F-XBT:EUR': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-XBT:EUR',
                                    'pair': 'XBT/EUR',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'EUR',
                                    'unrealized_funding': 0.0},
                      'F-XBT:USD': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-XBT:USD',
                                    'pair': 'XBT/USD',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'XBT',
                                    'unrealized_funding': 0.0},
                      'F-XRP:USD': {'available': 0.0,
                                    'balance': 0.0,
                                    'initial_margin': 0.0,
                                    'maintenance_margin': 0.0,
                                    'name': 'F-XRP:USD',
                                    'pair': 'XRP/USD',
                                    'pnl': 0.0,
                                    'portfolio_value': 0.0,
                                    'portfolio_value_usd': 0.0,
                                    'unit': 'XRP',
                                    'unrealized_funding': 0.0}},
          'holding': {'BCH': 35.6328392246,
                      'ETH': 3.0445849013,
                      'EUR': 4699.2481203008,
                      'GBP': 4139.4840547074,
                      'LTC': 52.443885043,
                      'USD': 0.0,
                      'USDC': 5002.0008003201,
                      'USDT': 5000.500050005,
                      'XBT': 0.2071703309,
                      'XRP': 12841.2563885251},
          'seq': 0,
          'timestamp': 1692961382880},
 'timestamp': 1692961382880}
```

**Comment by carlosmiei** [comment]
@BaoziCDR oh ok you're right we're missing the load markets there, will add it asap. In the meantime, you can add `await exchange.loadMarkets()` before calling `watchBalance` as a workaround.
